### ISSUE TITLE
Validate the lr scheduler

### ISSUE THREAD

**Issue by carbonscott** [carbonscott/peaknet#7]
Inconsistency in the learning rate resumption.  Learning rate at iteration 5 of run n+1 isn't the same as the one at iteration 95 of run n.

### Run n

```
rank=0 | logevent=LOSS:TRAIN | iteration=87 | segment=9600-15177 | learning_rate=0.000294 | grad_norm=0.895882 | mean_train_loss=0.013059 | mean_mse_loss=0.029730 | mean_kl_loss=0.002891 | mean_focal_loss=0.000054 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.697 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=88 | segment=9600-15177 | learning_rate=0.000294 | grad_norm=0.943161 | mean_train_loss=0.013118 | mean_mse_loss=0.029540 | mean_kl_loss=0.003226 | mean_focal_loss=0.000057 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=89 | segment=9600-15177 | learning_rate=0.000294 | grad_norm=0.557374 | mean_train_loss=0.012179 | mean_mse_loss=0.027533 | mean_kl_loss=0.002887 | mean_focal_loss=0.000055 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=90 | segment=9600-15177 | learning_rate=0.000294 | grad_norm=0.651592 | mean_train_loss=0.012185 | mean_mse_loss=0.027659 | mean_kl_loss=0.002779 | mean_focal_loss=0.000050 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=4
rank=0 | logevent=LOSS:TRAIN | iteration=91 | segment=0-9600 | learning_rate=0.000294 | grad_norm=1.079075 | mean_train_loss=0.013508 | mean_mse_loss=0.030822 | mean_kl_loss=0.002923 | mean_focal_loss=0.000054 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.697 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=92 | segment=0-9600 | learning_rate=0.000294 | grad_norm=0.768269 | mean_train_loss=0.012368 | mean_mse_loss=0.028078 | mean_kl_loss=0.002815 | mean_focal_loss=0.000056 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=93 | segment=0-9600 | learning_rate=0.000294 | grad_norm=0.699767 | mean_train_loss=0.012147 | mean_mse_loss=0.027410 | mean_kl_loss=0.002930 | mean_focal_loss=0.000056 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=94 | segment=0-9600 | learning_rate=0.000294 | grad_norm=0.882056 | mean_train_loss=0.012492 | mean_mse_loss=0.028056 | mean_kl_loss=0.003146 | mean_focal_loss=0.000056 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.697 | grad_nosync_counter=10
*** rank=0 | logevent=LOSS:TRAIN | iteration=95 | segment=0-9600 | learning_rate=0.000294 | grad_norm=0.615322 | mean_train_loss=0.011874 | mean_mse_loss=0.026816 | mean_kl_loss=0.002841 | mean_focal_loss=0.000056 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=96 | segment=0-9600 | learning_rate=0.000294 | grad_norm=0.159981 | mean_train_loss=0.011364 | mean_mse_loss=0.025604 | mean_kl_loss=0.002779 | mean_focal_loss=0.000054 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=97 | segment=9600-15177 | learning_rate=0.000294 | grad_norm=0.526744 | mean_train_loss=0.011311 | mean_mse_loss=0.025504 | mean_kl_loss=0.002747 | mean_focal_loss=0.000053 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.697 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=98 | segment=9600-15177 | learning_rate=0.000294 | grad_norm=0.601726 | mean_train_loss=0.011469 | mean_mse_loss=0.025921 | mean_kl_loss=0.002725 | mean_focal_loss=0.000053 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.697 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=99 | segment=9600-15177 | learning_rate=0.000294 | grad_norm=0.251167 | mean_train_loss=0.010817 | mean_mse_loss=0.024348 | mean_kl_loss=0.002669 | mean_focal_loss=0.000052 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=100 | segment=9600-15177 | learning_rate=0.000294 | grad_norm=0.392758 | mean_train_loss=0.011139 | mean_mse_loss=0.025069 | mean_kl_loss=0.002752 | mean_focal_loss=0.000056 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.701 | grad_nosync_counter=4
```

### Run n+1

```
rank=0 | logevent=LOSS:TRAIN | iteration=1 | segment=0-9600 | learning_rate=0.000294 | grad_norm=0.384464 | mean_train_loss=0.011113 | mean_mse_loss=0.025057 | mean_kl_loss=0.002699 | mean_focal_loss=0.000053 | tokens_per_sec=2.6e+06 | mfu_per_iteration=0.689 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=2 | segment=0-9600 | learning_rate=0.000294 | grad_norm=0.628329 | mean_train_loss=0.011459 | mean_mse_loss=0.025906 | mean_kl_loss=0.002714 | mean_focal_loss=0.000054 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=3 | segment=0-9600 | learning_rate=0.000294 | grad_norm=0.603688 | mean_train_loss=0.011422 | mean_mse_loss=0.025825 | mean_kl_loss=0.002704 | mean_focal_loss=0.000054 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.697 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=4 | segment=0-9600 | learning_rate=0.000294 | grad_norm=0.795651 | mean_train_loss=0.011615 | mean_mse_loss=0.026226 | mean_kl_loss=0.002784 | mean_focal_loss=0.000055 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
*** rank=0 | logevent=LOSS:TRAIN | iteration=5 | segment=0-9600 | learning_rate=0.000293 | grad_norm=1.485929 | mean_train_loss=0.013528 | mean_mse_loss=0.030856 | mean_kl_loss=0.002937 | mean_focal_loss=0.000053 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=6 | segment=0-9600 | learning_rate=0.000293 | grad_norm=0.475933 | mean_train_loss=0.010930 | mean_mse_loss=0.024606 | mean_kl_loss=0.002693 | mean_focal_loss=0.000054 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=7 | segment=9600-15177 | learning_rate=0.000293 | grad_norm=0.475809 | mean_train_loss=0.010714 | mean_mse_loss=0.024087 | mean_kl_loss=0.002672 | mean_focal_loss=0.000053 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.697 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=8 | segment=9600-15177 | learning_rate=0.000293 | grad_norm=1.282681 | mean_train_loss=0.012444 | mean_mse_loss=0.028255 | mean_kl_loss=0.002828 | mean_focal_loss=0.000053 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=9 | segment=9600-15177 | learning_rate=0.000293 | grad_norm=0.998206 | mean_train_loss=0.011460 | mean_mse_loss=0.025949 | mean_kl_loss=0.002674 | mean_focal_loss=0.000052 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=10 | segment=9600-15177 | learning_rate=0.000293 | grad_norm=0.931817 | mean_train_loss=0.011408 | mean_mse_loss=0.025909 | mean_kl_loss=0.002585 | mean_focal_loss=0.000053 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.699 | grad_nosync_counter=4
rank=0 | logevent=LOSS:TRAIN | iteration=11 | segment=0-9600 | learning_rate=0.000293 | grad_norm=0.990194 | mean_train_loss=0.011564 | mean_mse_loss=0.026316 | mean_kl_loss=0.002569 | mean_focal_loss=0.000052 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.697 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=12 | segment=0-9600 | learning_rate=0.000293 | grad_norm=0.898148 | mean_train_loss=0.011231 | mean_mse_loss=0.025406 | mean_kl_loss=0.002645 | mean_focal_loss=0.000053 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=13 | segment=0-9600 | learning_rate=0.000293 | grad_norm=0.503397 | mean_train_loss=0.010882 | mean_mse_loss=0.024526 | mean_kl_loss=0.002654 | mean_focal_loss=0.000054 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=14 | segment=0-9600 | learning_rate=0.000293 | grad_norm=0.135260 | mean_train_loss=0.010292 | mean_mse_loss=0.023141 | mean_kl_loss=0.002562 | mean_focal_loss=0.000052 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.697 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=15 | segment=0-9600 | learning_rate=0.000293 | grad_norm=0.619225 | mean_train_loss=0.010640 | mean_mse_loss=0.023998 | mean_kl_loss=0.002576 | mean_focal_loss=0.000052 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=16 | segment=0-9600 | learning_rate=0.000293 | grad_norm=0.811145 | mean_train_loss=0.011100 | mean_mse_loss=0.025147 | mean_kl_loss=0.002578 | mean_focal_loss=0.000051 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=17 | segment=9600-15177 | learning_rate=0.000293 | grad_norm=0.656559 | mean_train_loss=0.010573 | mean_mse_loss=0.023828 | mean_kl_loss=0.002577 | mean_focal_loss=0.000053 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.697 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=18 | segment=9600-15177 | learning_rate=0.000293 | grad_norm=0.335843 | mean_train_loss=0.009987 | mean_mse_loss=0.022475 | mean_kl_loss=0.002466 | mean_focal_loss=0.000051 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=19 | segment=9600-15177 | learning_rate=0.000293 | grad_norm=0.433668 | mean_train_loss=0.010087 | mean_mse_loss=0.022708 | mean_kl_loss=0.002485 | mean_focal_loss=0.000050 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=20 | segment=9600-15177 | learning_rate=0.000293 | grad_norm=0.813213 | mean_train_loss=0.010547 | mean_mse_loss=0.023637 | mean_kl_loss=0.002703 | mean_focal_loss=0.000053 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.701 | grad_nosync_counter=4
rank=0 | logevent=LOSS:TRAIN | iteration=21 | segment=0-9600 | learning_rate=0.000293 | grad_norm=1.144834 | mean_train_loss=0.011354 | mean_mse_loss=0.025655 | mean_kl_loss=0.002704 | mean_focal_loss=0.000050 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.697 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=22 | segment=0-9600 | learning_rate=0.000293 | grad_norm=0.918951 | mean_train_loss=0.010883 | mean_mse_loss=0.024473 | mean_kl_loss=0.002708 | mean_focal_loss=0.000051 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=23 | segment=0-9600 | learning_rate=0.000293 | grad_norm=1.099586 | mean_train_loss=0.011482 | mean_mse_loss=0.026087 | mean_kl_loss=0.002592 | mean_focal_loss=0.000053 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=24 | segment=0-9600 | learning_rate=0.000293 | grad_norm=1.279673 | mean_train_loss=0.011858 | mean_mse_loss=0.027131 | mean_kl_loss=0.002489 | mean_focal_loss=0.000050 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=25 | segment=0-9600 | learning_rate=0.000293 | grad_norm=0.460272 | mean_train_loss=0.010112 | mean_mse_loss=0.022733 | mean_kl_loss=0.002521 | mean_focal_loss=0.000053 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
rank=0 | logevent=LOSS:TRAIN | iteration=26 | segment=0-9600 | learning_rate=0.000293 | grad_norm=0.694068 | mean_train_loss=0.010430 | mean_mse_loss=0.023542 | mean_kl_loss=0.002509 | mean_focal_loss=0.000050 | tokens_per_sec=2.7e+06 | mfu_per_iteration=0.698 | grad_nosync_counter=10
```
